src/components/mixins/ProfilesMixin.vue

```diff
@@ -0,0 +1,141 @@
+<script lang='ts'>
+import Vue from 'vue';
+import Component from 'vue-class-component';
+
+import ZipProvider from "../../providers/generic/zip/ZipProvider";
+import path from "path";
+import Profile from "../../model/Profile";
+import ThunderstoreMod from "../../model/ThunderstoreMod";
+import ThunderstoreVersion from "../../model/ThunderstoreVersion";
+import R2Error from "../../model/errors/R2Error";
+import ManifestV2 from "../../model/ManifestV2";
+import ProfileInstallerProvider from "../../providers/ror2/installing/ProfileInstallerProvider";
+import ProfileModList from "../../r2mm/mods/ProfileModList";
+import sanitize from "sanitize-filename";
+import ManagerInformation from "../../_managerinf/ManagerInformation";
+import FsProvider from "../../providers/generic/file/FsProvider";
+import * as yaml from "yaml";
+import ExportFormat from "../../model/exports/ExportFormat";
+import ExportMod from "../../model/exports/ExportMod";
+import VersionNumber from "../../model/VersionNumber";
+
+@Component
+export default class ProfilesMixin extends Vue {
+    get profileList(): string[] {
+        return this.$store.state.profiles.profileList;
+    }
+    get appName(): string {
+        return ManagerInformation.APP_NAME;
+    }
+    get activeProfileName(): string {
+        return this.$store.getters['profile/activeProfileName'];
+    }
+    get activeProfile(): Profile {
+        return this.$store.getters['profile/activeProfile'];
+    }
+
+    doesProfileExist(nameToCheck: string): boolean {
+        if ((nameToCheck.match(new RegExp('^([a-zA-Z0-9])(\\s|[a-zA-Z0-9]|_|-|[.])*$'))) === null) {
+            return true;
+        }
+        const safe: string = this.makeProfileNameSafe(nameToCheck);
+        return (this.profileList.some(function (profile: string) {
+            return profile.toLowerCase() === safe.toLowerCase()
+        }));
+    }
+
+    makeProfileNameSafe(nameToSanitize: string): string {
+        return sanitize(nameToSanitize);
+    }
+
+    async setSelectedProfile(profileName: string, prewarmCache = true) {
```

Comment:
As far as I can tell, even when checking the next PR in the stack, this method and all below it in this file, are only used by Profiles.vue. If this is the case, what is the rationale of moving them into the mixin? Seems we would get the cons (having the code split into multiple places) without the pros (code reusability).

This also applies to `appName` getter.